x-pack/plugins/observability_solution/observability/public/pages/alert_details/components/header_actions.tsx

```diff
@@ -22,23 +22,37 @@ import {
 } from '@elastic/eui';
 import {
   AlertStatus,
+  AlertConsumers,
   ALERT_RULE_UUID,
   ALERT_STATUS_ACTIVE,
   ALERT_UUID,
+  ALERT_RULE_PRODUCER,
+  ALERT_RULE_CONSUMER,
 } from '@kbn/rule-data-utils';
 
 import { useKibana } from '../../../utils/kibana_react';
 import { useFetchRule } from '../../../hooks/use_fetch_rule';
 import type { TopAlert } from '../../../typings/alerts';
 import { paths } from '../../../../common/locators/paths';
 import { useBulkUntrackAlerts } from '../hooks/use_bulk_untrack_alerts';
+import { observabilityFeatureId } from '../../../../common';
 
 export interface HeaderActionsProps {
   alert: TopAlert | null;
   alertStatus?: AlertStatus;
   onUntrackAlert: () => void;
 }
 
+const getAlertFeatureId = (alert: TopAlert): AlertConsumers => {
+  if (alert.fields[ALERT_RULE_PRODUCER] !== AlertConsumers.STACK_ALERTS) {
```

Comment:
In theory this should never happen, after a discussion with @XavierM, we decided we should try to derive the feature ID from the rule, instead of hardcoding `observability` for the feature ID. To make sure we do not pass in `stack_alerts` for the o11y use case.

Reply:
This logic is a bit hard to understand and seems fragile. When talking to @umbopepato , this seemed similar to what we are doing [here](https://github.com/elastic/kibana/pull/175143/files#diff-d2e5347263c3c11758a0753573ed718058e00808f2b3b37c08dbb21c4f824bd0R150-R154). I wonder if we can maybe have a utility with documentation and proper testing and use it to calculate which featureId should be used for an alert in different places. 

@JiaweiWu @XavierM wdyt?